text/0000-allow-sharing-of-LWC-across-namespaces.md

```diff
@@ -0,0 +1,230 @@
+---
+title: Allow sharing of LWC across namespaces 
+status: DRAFTED
+created_at: 2020-03-29
+updated_at: 2020-04-21
+pr: (leave this empty until the PR is created)
+---
+
+# Allow sharing of LWC across namespaces
+
+## Summary
+
+There is a tight coupling (1:1) between namespace and Lighting Web Components (LWC). Developer can use isExposed flag to expose the component to the world or keep it private within the namespace.
+
+This RFC proposes functionality to allow sharing of LWC across namespaces.
+
+
+## Motivation
+
+Some teams have more than one namespaces and only desire to share their common components internally. However, it is not possible today, which forces such teams to make their component available globally. This is an undesired effect as it introduces maintenance burden.

```

Comment:
It's something that I was thinking about recently, why do teams own multiple namespaces? As far as I know, teams use multiple namespaces to have a structural breakdown between the different modules composing a feature or a set of features. This approach is used due to the flat nature of the LWC module structure.

IMHO, this proposal is attempting to solve the symptoms instead of addressing the core problem which is the lack of flexibility in the way modules can be structured into a single namespace. 

/cc @diervo @kevinv11n @caridy

Reply:
The specific use case here is to allow ISV to build the shared library (utility packages or common components) which could be shared internally across their customer.

Reply:
@shahprincea We need to clearly articulate the audience for this feature. Here you are talking about ISVs, who are external customers. However you also use the word "internal" a lot, which implies that the feature only applies to SFDC teams, viz., code running in core. Not on platform.

Reply:
For some random prior art on 'sharing internals', see: https://www.vogella.com/tutorials/OSGi/article.html#defining-provisional-api-via-the-x-internal-and-x-friends

Reply:
Copying my commend from the quip rfc:
Internal teams having multiple namespaces is a reality that we live in today, it is not a symptom, it is a diagnosis :). In my opinion, lwc flat structure isn't something that will be changing any time soon, meanwhile developer ergonomics is suffering. With that, this proposal not only allows teams to logically organize their code base, but it also enables fine-grained code re-use across other namespaces. Just yesterday, we've had a team [asking](https://gus.lightning.force.com/lightning/r/0D5B0000016c85yKAA/view) if they can share their library to a specific namespace that falls outside of theirs. 
In my opinion this proposal doesn't conflict with working out the details of LWC structure. However, I do agree that all the use cases must be specified in the proposal for clarity.